sniffer/tools/sniffer.py

```diff
@@ -135,25 +136,39 @@ def connect(argv):
 
     return conn
 
-def main(argv):
-    if len(argv) < 5:
-        print("Usage: %s serial tty baudrate channel [outfile]\n"
-              "       %s socket host port channel [outfile]" % (argv[0], argv[0]),
-              file=sys.stderr)
-        print("       channel = 11-26", file=sys.stderr)
-        sys.exit(2)
-
-    conn = connect(argv)
+def main():
+    p = argparse.ArgumentParser()
+    sp = p.add_subparsers(dest="conn_type")
+    serial_p = sp.add_parser("serial",
+                             help="Parse output of sniffer application "
+                                  "connected via serial line")
+    serial_p.add_argument("tty", type=str,
+                          help="Serial port to board with sniffer application")
+    serial_p.add_argument("-b", "--baudrate", type=int, default=112500,
+                          nargs="?", help="Baudrate of the serial port")
+    socket_p = sp.add_parser("socket",
+                             help="Parse output of a TCP-connected sniffer "
+                                  "application")
+    socket_p.add_argument("host", type=str,
+                          help="Host of the TCP-based sniffer application")
+    socket_p.add_argument("port", type=int,
+                          help="Port of the TCP-based sniffer application")
+    p.add_argument("channel", type=int, help="Channel to sniff on")
+    p.add_argument("outfile", type=str, default=None, nargs="?",
+                   help="PCAP file to output to")
```

Comment:
There is `argparse.FileType` to handle file arguments in python.

Reply:
This change is part of #37. (sorry, did not make clear this PR depends on that)

Reply:
But I'll have a look!

Reply:
Ah, that explains why my comment didn't show up at #37, I meant to place it there :)